features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/attachments/video/VideoMetadataExtractor.kt

```diff
@@ -38,13 +39,18 @@ class DefaultVideoMetadataExtractor @AssistedInject constructor(
         override fun create(uri: Uri): DefaultVideoMetadataExtractor
     }
 
+    private val isDataSourceReady = AtomicBoolean(false)
+
     private val mediaMetadataRetriever = MediaMetadataRetriever()
 
-    init {
-        mediaMetadataRetriever.setDataSource(context, uri)
+    private fun ensureDataSourceIsReady() {
```

Comment:
Maybe another simpler fix would be to lazy create the MediaMetaDataRetriever?
```kotlin
    private val mediaMetadataRetriever by lazy {
        MediaMetadataRetriever().also {
            it.setDataSource(context, uri)
        }
    }
```
Or the crash do not occur anymore since the exception is caught by `runCatchingExceptions`? In this case I am not sure that my suggestion will work the same way.

Reply:
Well, it works in the sense that it doesn't crash anymore, but if we called one of those methods by mistake, the exception wouldn't be caught by `runCatchingExceptions`.